qemu/tests/qemu_guest_agent.py

```diff
@@ -60,7 +60,8 @@ def cleanup(self, test, params, env):
 
     def execute(self, test, params, env):
         self.initialize(test, params, env)
-        self.setup(test, params, env)
+        if params.get("gagent_serial_type"):
+            self.setup(test, params, env)
```

Comment:
I think it's not good to modify it in the base class.
You can refer to "-check_fstrim" test case, it set  start_vm = no and then boot up guest in your sub function.

Reply:
I don't think the added if will influence other cases, because the "gagent_serial_type" always exists for other cases, self.setup will be execucated as before. But you remind me, for L235-L257 and L1040-L1063, I should add another judgement if start_vm == "no".

Reply:
I still think you should not add this condition in this base function, in setup() steps ,there is  if (not self.vm) and (start_vm == "yes"): condition, you just need set start_vm = no in cfg.